apps/discord-bot-internal/src/interaction-modules/roles.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  AutocompleteInteraction,
+  ChatInputCommandInteraction,
+  Colors,
+  GuildMember,
+  SlashCommandBuilder
+} from 'discord.js';
+import { InteractionModule } from '../types/interaction-module';
+import {
+  isAdmin,
+  isAdminBotChannel,
+  isModeratorOrHigher,
+  replyDescriptionEmbed
+} from '../utils';
+import { MomentumColor } from '../momentum-color';
+import { config } from '../config';
+
+export class RolesModule implements InteractionModule {
+  userFilter = isModeratorOrHigher;
+  channelFilter = isAdminBotChannel;
+  commandBuilder = new SlashCommandBuilder()
+    .setName('role')
+    .setDescription('Roles')
+    .addSubcommand((builder) =>
+      builder
+        .setName('give')
+        .setDescription('Give a role to a member')
+        .addUserOption((option) =>
+          option.setName('member').setDescription('member').setRequired(true)
+        )
+        .addStringOption((option) =>
+          option
+            .setName('role')
+            .setDescription('Role')
+            .setRequired(true)
+            .setAutocomplete(true)
+        )
+    )
+    .addSubcommand((builder) =>
+      builder
+        .setName('take')
+        .setDescription('Take a role from a member')
```

Comment:
Not a fan of this naming, wasn't 100% sure what "take" meant at first. I think it'd be better to use "add"/"remove" terminology for this, then something more verbose for the "giveable" stuff, maybe "makeGiveable" / "unmakeGiveable"?